File: android/src/main/kotlin/com/unico_check/config/UnicoTheme.kt
package com.unico_check.config

import com.acesso.acessobio_android.onboarding.IAcessoBioTheme
import com.unico_check.constants.MethodConstants.UNICO_THEME
import io.flutter.plugin.common.MethodCall

data class UnicoTheme(private val theme: Map<Any, Any>) : IAcessoBioTheme {


    override fun getColorBackground(): Any {
        return theme["colorBackground"] ?: 0
    }

    override fun getColorBoxMessage(): Any {
        return theme["colorBoxMessage"] ?: 0
    }

    override fun getColorTextMessage(): Any {
        return theme["colorTextMessage"] ?: 0
    }

    override fun getColorBackgroundPopupError(): Any {
        return theme["colorBackgroundPopupError"] ?: 0
    }

    override fun getColorTextPopupError(): Any {
        return theme["colorTextPopupError"] ?: 0
    }

    override fun getColorBackgroundButtonPopupError(): Any {
        return theme["colorBackgroundButtonPopupError"] ?: 0
    }

    override fun getColorTextButtonPopupError(): Any {
        return theme["colorTextButtonPopupError"] ?: 0
    }

    override fun getColorBackgroundTakePictureButton(): Any {
        return theme["colorBackgroundTakePictureButton"] ?: 0
    }

    override fun getColorIconTakePictureButton(): Any {
        return theme["colorIconTakePictureButton"] ?: 0
    }

    override fun getColorBackgroundBottomDocument(): Any {
        return theme["colorBackgroundBottomDocument"] ?: 0
    }

    override fun getColorTextBottomDocument(): Any {
        return theme["colorTextBottomDocument"] ?: 0
    }

    override fun getColorSilhouetteSuccess(): Any {
        return theme["colorSilhouetteSuccess"] ?: 0
    }

    override fun getColorSilhouetteError(): Any {
        return theme["colorSilhouetteError"] ?: 0
    }

    override fun getColorSilhouetteNeutral(): Any {
        return theme["colorSilhouetteNeutral"] ?: 0
    }
}

File: ios/Classes/config/UnicoTheme.swift
//
//  UnicoTheme.swift
//  Pods
//
//  Created by Lucas Diniz Silva on 26/08/21.
//

import Foundation

class UnicoTheme: NSObject, AcessoBioThemeDelegate {
    private var argument: Dictionary<String, Any>?
    
    init(argument: Dictionary<String, Any>){
        self.argument = argument[MethodConstants.UNICO_THEME.rawValue] as? Dictionary<String, Any>
    }
    
    func getColorBackground() -> Any? {
        return self.argument?["colorBackground"] as? String
    }
    
    func getColorBoxMessage() -> Any? {
        return self.argument?["colorBoxMessage"] as? String
    }
    
    func getColorTextMessage() -> Any? {
        return self.argument?["colorTextMessage"] as? String
    }
    
    func getColorBackgroundPopupError() -> Any? {
        return self.argument?["colorBackgroundPopupError"] as? String
    }
    
    func getColorTextPopupError() -> Any? {
        return self.argument?["colorTextPopupError"] as? String
    }
    
    func getColorBackgroundButtonPopupError() -> Any? {
        return self.argument?["colorBackgroundButtonPopupError"] as? String
    }
    
    func getColorTextButtonPopupError() -> Any? {
        return self.argument?["colorTextButtonPopupError"] as? String
    }
    
    func getColorBackgroundTakePictureButton() -> Any? {
        return self.argument?["colorBackgroundTakePictureButton"] as? String
    }
    
    func getColorIconTakePictureButton() -> Any? {
        return self.argument?["colorIconTakePictureButton"] as? String
    }
    
    func getColorBackgroundBottomDocument() -> Any? {
        return self.argument?["colorBackgroundBottomDocument"] as? String
    }
    
    func getColorTextBottomDocument() -> Any? {
        return self.argument?["colorTextBottomDocument"] as? String
    }
    
    func getColorSilhouetteSuccess() -> Any? {
        return self.argument?["colorSilhouetteSuccess"] as? String
    }
    
    func getColorSilhouetteError() -> Any? {
        return self.argument?["colorSilhouetteError"] as? String
    }
    
    func getColorSilhouetteNeutral() -> Any? {
        return self.argument?["colorSilhouetteNeutral"] as? String
    }

    func getProgressBarColor() -> Any? {
        return self.argument?["colorProgressBar"] as? String
    }

    func getCancelButtonIconColor() -> Any? {
        return self.argument?["colorCancelButtonIcon"] as? String
    }
}

